scripts/poolparty.lic

```diff
@@ -56,183 +56,151 @@
+  def id
+    @item.id
+  end
 
-    if XMLData.room_title == '[Pinefar, Depository]'
-      if GameObj.npcs.any? {|npc| npc.noun == 'banker'}
-        fput "ask banker for #{[withdraw_amount.to_i, 20].max} silvers"
-        # The banker nods and says, "Alright, here ye go.  Ye understand I be takin' a little more than that from ye account in the 'Mule.  I don't works for free!"
-        # The banker looks at you suspiciously and says, "Hmm, I don't think ye be havin' enough in ye account to cover that and my fee.  Ye tryin' to pull one over on me?"
-      else
-        if (icemule_bank = Room.list.find {|room| room.location == 'Icemule Trace' and room.tags.include?('bank')})
-          Script.run('go2', icemule_bank.id.to_s)
-          result = dothistimeout "withdraw #{withdraw_amount} silvers", 1, /debt collector/
-          if result =~ /debt collector/
-            fput "withdraw #{withdraw_amount} silvers"
-          end
-        else
-          echo 'waiting for banker...'
-          wait_until {GameObj.npcs.any? {|npc| npc.noun == 'banker'}}
-          fput "ask banker for #{[withdraw_amount.to_i, 20].max} silvers"
-        end
-      end
-    else
-      result = dothistimeout "withdraw #{withdraw_amount} silvers", 1, /debt collector/
-      if result =~ /debt collector/
-        fput "withdraw #{withdraw_amount} silvers"
-      end
-    end
+  def noun
+    @item.noun
   end
 
-  def self.check_for_disk
+  def name
+    @item.name
+  end
 
-    disk = nil
-    notified = false
-    25.times {
-      break if (disk = GameObj[/^.*#{Char.name} (?:disk|coffin)$/])
-      sleep 0.2
-      unless notified
+  def type
+    @item.type
+  end
 
-        Utilities::send_formatted("  `Waiting on your disk to arrive`")
-        notified = true
-      end
-    }
+  def is_valid?
+    !@item.id.nil?
+  end
+
+end
 
-    disk
+class Container < LichObject
+  def initialize(container)
+    super container
   end
 
-  def self.deposit_silver(amount)
-    Script.run('go2', 'bank --disable-confirm')
-    if XMLData.room_title == '[Pinefar, Depository]'
-      if GameObj.npcs.any? {|npc| npc.noun == 'banker'}
-        fput "give banker #{amount} silvers"
-      end
-    else
-      fput "deposit #{amount}"
-    end
+  def open
+    fput "open ##{@item.id}"
   end
+end
 
-  #Thank you foreach for all this lovely helpful code
-  def self.find_sack(base = 'lootsack')
-    # Adapted from ;loot
-    sack = nil
-    var = "#{base}"
-    name = Vars[var].strip
-    return sack unless name.length > 0
-    if name =~ /^my\s+(.*)$/i
-      name = $1
-    end
-    pattern = Regexp::escape(name)
-    sack = nil
-    [/\b#{pattern}$/i, /\b#{pattern.sub(' ', '.*')}$/i, /\b#{pattern.sub(' ', '.*')}/i].each {|pattern|
-      sack = GameObj.inv.find {|obj| obj.name =~ pattern}
-      break if sack
-    }
-    if sack
-      sack
-    else
-      echo "warning: failed to find #{var} '#{name}'"
-    end
+class Box < LichObject

```

Comment:
This can break other scripts, please scope this top-level `Box` class to the `PoolParty` module.

Reply:
Fixed in commit